File: Chapter 02/AdapterDesignPatternExample.java
interface TargetExample {
    void request();
}

e.
class AdapteeExample {
    void specificRequest() {
        System.out.println("This is from the adaptee's specific request.");
    }
}

class AdapterExample implements TargetExample {
    private AdapteeExample adaptee;

    public AdapterExample(AdapteeExample adaptee) {
        this.adaptee = adaptee;
    }

    public void request() {
        System.out.println("This is the adapter's request");
        adaptee.specificRequest();
    }
}

public class AdapterDesingPatternExample {
    public static void main(String[] args) {
        AdapteeExample adaptee = new AdapteeExample();
        TargetExample adapter = new AdapterExample(adaptee);

        adapter.request();
    }
}

File: Chapter 02/AdapterDesignPatternExample.py
class TargetExample:
    def request(self):
        pass
class AdapteeExample:
    def specific_request(self):
        print("This is the adaptee's specific request.")
class AdapterExample(TargetExample):
    def __init__(self, adaptee):
        self.adaptee = adaptee
    def request(self):
        print("This is the adapter's request")
        self.adaptee.specific_request()
if __name__ == "__main__":
    adaptee = AdapteeExample()
    adapter = AdapterExample(adaptee)

    adapter.request()
